### PR TITLE
[bull] add missing `options` parameter in backoffStrategies callback

### PR DESCRIPTION
**Pull request by paulintrognon**
I am really not a fan of using an `any` here, but I could not achieve to infer type of options from the `BackoffOptions` type.

---

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

- [X] Provide a URL to documentation or source code which provides context for the suggested changes: [github.com/OptimalBits/bull/lib/backoffs.js](https://github.com/OptimalBits/bull/blob/develop/lib/backoffs.js#L53)
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## bull ([<kbd>unpkg</kbd>](https://unpkg.com/browse/bull@latest/))
was missing the following properties:
1. ErrorMessages
1. Job
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against db2d368c403aa3d38ad8fb71e90d282c2e994091
</p>

**Comment by danmana** [comment]
Note, the next major version of Bull is written in typescript.  
Maybe you can get some idea of the options from: https://github.com/taskforcesh/bullmq/blob/master/src/interfaces/backoff-options.ts

**Comment by paulintrognon** [comment]
> Note, the next major version of Bull is written in typescript. Maybe you can get some idea of the options from: https://github.com/taskforcesh/bullmq/blob/master/src/interfaces/backoff-options.ts

Interesting, but this new version of bull you are pointing out does not implement custom options... It seems the new version does not allow passing custom options for custom backoff strategies.

`bull`: https://github.com/OptimalBits/bull/blob/develop/lib/backoffs.js#L53
vs
`bullmq`: https://github.com/taskforcesh/bullmq/blob/master/src/classes/backoffs.ts#L54

File: types/bull/index.d.ts [hunk 1]
```diff
@@ -111,7 +111,7 @@ declare namespace Bull {
      * Define a custom backoff strategy
      */
     backoffStrategies?: {
-      [key: string]: (attemptsMade: number, err: Error) => number;
+      [key: string]: (attemptsMade: number, err: Error, options?: any) => number;
```
Comment (bondz) [inline types/bull/index.d.ts:114]:
```suggestion
      [key: string]: (attemptsMade: number, err: Error, strategyOptions?: any) => number;
```

File: types/bull/index.d.ts [hunk 2]
```diff
@@ -331,6 +331,11 @@ declare namespace Bull {
      * Backoff delay, in milliseconds
      */
     delay?: number | undefined;
+
+    /**
+     * Options for custom strategies
+     */
+    options?: any;
```
Comment (bondz) [inline types/bull/index.d.ts:338]:
```suggestion
    strategyOptions?: any;
```

**Review by bondz — APPROVED** [review]
I think it should be `strategyOptions` as in the code.

**Comment by paulintrognon** [comment]
> I think it should be `strategyOptions` as in the code.

You're right, I changed it.

**Review by bondz — APPROVED** [review]

**Comment by paulintrognon** [comment]
Ready to merge

**Pull request merged**